test/e2e-common.sh

```diff
@@ -122,6 +122,8 @@ function build_knative_from_source() {
   local FULL_OUTPUT="$( \
       source $(dirname $0)/../hack/generate-yamls.sh ${REPO_ROOT_DIR} ${YAML_LIST} ; \
       set | grep _YAML=/)"
+  # Fail if generate-yamls.sh failed
+  (( $? )) && return 1
```

Comment:
Good catch, Chao. However, I don't think this will work:

* FULL_OUTPUT is a combination of 2 commands, one being a pipe, and grep will only fail if no _YAML=/ is found
* if FULL_OUTPUT is corrupt, a clear error message would be thrown in 129
* the environment variables are actually set, as you can see in the logs
* generate-yamls.sh should exit with an error code if ko fails to do something (https://github.com/knative/serving/blob/master/hack/generate-yamls.sh#L38), but that didn't happen (the logs show that other images were built), so it seems that the root cause is ko not failing properly when uploading the image to the registry

Reply:
That's right, it was the wrong fix. I have checked `ko` implementation, and it did exit correctly. The real problem is that `local` swallows exit status, so updated with new fix, this should work